src/main/java/ch/njol/skript/conditions/CondCompare.java

```diff
@@ -160,57 +162,55 @@ public static String f(final Expression<?> e) {
 	
 	@SuppressWarnings({"unchecked"})
 	private boolean init(String expr) {
-		final RetainingLogHandler log = SkriptLogger.startRetainingLog();
+		RetainingLogHandler log = SkriptLogger.startRetainingLog();
 		Expression<?> third = this.third;
 		try {
 			if (first.getReturnType() == Object.class) {
-				final Expression<?> e = first.getConvertedExpression(Object.class);
-				if (e == null) {
+				Expression<?> expression = null;
+				if (first instanceof UnparsedLiteral)
+					expression = attemptReconstruct((UnparsedLiteral) first, second);
+				if (expression == null)
+					expression = first.getConvertedExpression(Object.class);
+				if (expression == null) {
 					log.printErrors();
 					return false;
 				}
-				first = e;
+				first = expression;
 			}
 			if (second.getReturnType() == Object.class) {
-				final Expression<?> e = second.getConvertedExpression(Object.class);
-				if (e == null) {
+				Expression<?> expression = null;
+				if (second instanceof UnparsedLiteral)
+					expression = attemptReconstruct((UnparsedLiteral) second, first);
+				if (expression == null)
+					expression = second.getConvertedExpression(Object.class);
+				if (expression == null) {
 					log.printErrors();
 					return false;
 				}
-				second = e;
+				second = expression;
 			}
 			if (third != null && third.getReturnType() == Object.class) {
-				final Expression<?> e = third.getConvertedExpression(Object.class);
-				if (e == null) {
+				Expression<?> expression = null;
+				if (third instanceof UnparsedLiteral)
+					expression = attemptReconstruct((UnparsedLiteral) third, first);
+				if (expression == null)
+					expression = third.getConvertedExpression(Object.class);
+				if (expression == null) {
 					log.printErrors();
 					return false;
 				}
-				this.third = third = e;
+				this.third = third = expression;
 			}
 			log.printLog();
 		} finally {
 			log.stop();
 		}
-		final Class<?> f = first.getReturnType(), s = third == null ? second.getReturnType() : Utils.getSuperType(second.getReturnType(), third.getReturnType());
+		Class<?> f = first.getReturnType(), s = third == null ? second.getReturnType() : Utils.getSuperType(second.getReturnType(), third.getReturnType());
```

Comment:
Would be nice to have these renamed
```suggestion
		Class<?> firstReturnType = first.getReturnType(), ssecondReturnType = third == null ? second.getReturnType() : Utils.getSuperType(second.getReturnType(), third.getReturnType());
```